dataprofiler/profilers/histogram_utils.py

```diff
@@ -1,19 +1,241 @@
+
+    Function follows the numpy implementation within:
+    https://github.com/numpy/numpy/blob/main/numpy/lib/histograms.py
+
+    :param profile: Input data's data profile that is to be histogrammed
+    :type profile: NumericStatsMixin
+
+    :return: An estimate of the optimal bin width for the given data.
+    """
+    dataset_size = _get_dataset_size_from_profile(profile)
+    minimum = _get_minimum_from_profile(profile)
+    maximum = _get_maximum_from_profile(profile)
+
+    return _ptp(maximum, minimum) / np.sqrt(dataset_size)
+
+
+def _calc_fd_bin_width_from_profile(profile):
+    """
+    Execute Freedman-Diaconis histogram binning reworked to use profiles.
+
+    Function follows the numpy implementation within:
+    https://github.com/numpy/numpy/blob/main/numpy/lib/histograms.py
+
+    :param profile: Input data's data profile that is to be histogrammed
+    :type profile: NumericStatsMixin
+
+    :return: An estimate of the optimal bin width for the given data.
+    """
+    iqr = np.subtract(profile._get_percentile([75]), profile._get_percentile([25]))
+    dataset_size = _get_dataset_size_from_profile(profile)
+
+    return 2.0 * iqr * dataset_size ** (-1.0 / 3.0)
+
+
+def _calc_auto_bin_width_from_profile(profile):
+    """
+    Histogram bin estimator that uses Freedman-Diaconis and Sturges estimators.
+
+    Function follows the numpy implementation within:
+    https://github.com/numpy/numpy/blob/main/numpy/lib/histograms.py
+
+    :param profile: Input data's data profile that is to be histogrammed
+    :type profile: NumericStatsMixin
+
+    :return: An estimate of the optimal bin width for the given data.
+    """
+    fd_bw = _calc_fd_bin_width_from_profile(profile)
+    sturges_bw = _calc_sturges_bin_width_from_profile(profile)
+    if fd_bw:
+        return min(fd_bw, sturges_bw)
+    else:
+        # limited variance, so we return a len dependent bw estimator
+        return sturges_bw
+
+
+def _calc_scott_bin_width_from_profile(profile):
+    """
+    Scott histogram bin estimator reworked to use profiles.
+
+    Function follows the numpy implementation within:
+    https://github.com/numpy/numpy/blob/main/numpy/lib/histograms.py
+
+    :param profile: Input data's data profile that is to be histogrammed
+    :type profile: NumericStatsMixin
+
+    :return: An estimate of the optimal bin width for the given data.
+    """
+    dataset_size = _get_dataset_size_from_profile(profile)
+    std = profile.stddev
+
+    return (24.0 * np.pi**0.5 / dataset_size) ** (1.0 / 3.0) * std
+
+
+_hist_bin_width_selectors_for_profile = {
+    "auto": _calc_auto_bin_width_from_profile,
+    "doane": _calc_doane_bin_width_from_profile,
+    "fd": _calc_fd_bin_width_from_profile,
+    "rice": _calc_rice_bin_width_from_profile,
+    "scott": _calc_scott_bin_width_from_profile,
+    "sqrt": _calc_sqrt_bin_width_from_profile,
+    "sturges": _calc_sturges_bin_width_from_profile,
+}
+
+
 def _get_bin_edges(
     a: np.ndarray,
```

Comment:
think we might like a more descriptive variable name here?

Reply:
A future fix, not in code we changed and is actually how `numpy` also implements.

Reply:
good call